src/App.js

```diff
@@ -8,35 +10,28 @@ import Products from './components/Products';
 import ShoppingCart from './components/ShoppingCart';
 
 function App() {
-	const [products] = useState(data);
-	const [cart, setCart] = useState([]);
-
-	const addItem = item => {
-		setCart([...cart, item]);
-	};
-
-	return (
-		<div className="App">
-			<Navigation cart={cart} />
-
-			{/* Routes */}
-			<Route
-				exact
-				path="/"
-				render={() => (
-					<Products
-						products={products}
-						addItem={addItem}
-					/>
-				)}
-			/>
-
-			<Route
-				path="/cart"
-				render={() => <ShoppingCart cart={cart} />}
-			/>
-		</div>
-	);
+    const [products] = useState(data);
+
+    const [cart, setCart] = useState([]);
+
+    const addItem = item => {
+        setCart([...cart, item]);
+    };
+    return (
+        <ProductContext.Provider value={{products, addItem}}>
+            <CartContext.Provider value={cart}>
```

Comment:
I'm not 💯  on this but having a Provider provide a method that updates a different context seems messy. I will ask about this, I'd recommend looking into it as well.

Reply:
I was wrong 
![image](https://user-images.githubusercontent.com/20071094/65471860-b1d20c80-de25-11e9-8f08-8bba4980c3ee.png)